spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -157,7 +157,11 @@ private void assertEquals(String context, Object[] expectedRow, Object[] actualR
       Object actualValue = actualRow[col];
       if (expectedValue != null && expectedValue.getClass().isArray()) {
         String newContext = String.format("%s (nested col %d)", context, col + 1);
-        assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
+        if (expectedValue instanceof byte[]) {
+          Assert.assertArrayEquals(newContext, (byte[]) expectedValue, (byte[]) actualValue);
+        } else {
+          assertEquals(newContext, (Object[]) expectedValue, (Object[]) actualValue);
```

Comment:
Why does the `assertEquals` with `Object[]` not work for `byte[]`?

Reply:
The SQL query returns `byte[]` instead of `Byte[]`. The encapsulated type `Byte[]` can be converted to `Object[]`, but the conversion of the basic type `byte[]` will report an error, `[B cannot be cast to [Ljava.lang.Object`;